docs: `ChanUpgradeInit` and governance gating (#5347)
cosmos/ibc-go: docs/docs/01-ibc/06-channel-upgrades.md

Learn how to upgrade existing IBC channels.
:::

## Initializing a Channel Upgrade

A channel upgrade is initialised by submitting the `ChanUpgradeInit` message, which can be submitted only by the chain itself upon governance authorization. This message should specify an appropriate timeout window for the upgrade.

As part of the handling of the `ChanUpgradeInit` message, the application's callbacks `OnChanUpgradeInit` will be triggered as well.

After this message is handled successfully, the channel's upgrade sequence will be incremented. This upgrade sequence will serve as a nonce for the upgrade process to avoid error receipt proof collisions.

> ### Governance Gating on `ChanUpgradeInit`

Reviewer: ```suggestion
### Governance gating on `ChanUpgradeInit`
```
I believe titles should only have the first word capitalized

> A message signer who has been designated as the `authority` of the `IBCKeeper` sending the `MsgChanUpgradeInit`. If this proposal passes, the counterparty channels will upgrade by default.

Reviewer 2: There is a single channel version being upgraded per `MsgChanUpgradeInit` right?

```suggestion
A message signer who has been designated as the `authority` of the `IBCKeeper` sending the `MsgChanUpgradeInit`. If this proposal passes, the counterparty's channel will upgrade by default.
```

> A message signer who has been designated as the `authority` of the `IBCKeeper` sending the `MsgChanUpgradeInit`. If this proposal passes, the counterparty channels will upgrade by default.

Reviewer: ```suggestion
The message signer for `MsgChanUpgradeInit must be the address which has been designated as the `authority` of the `IBCKeeper` . If this proposal passes, the counterparty channels will upgrade by default.
```

> If chains want to initiate the upgrade of many channels, they will need to submit a governance proposal with multiple `MsgChanUpgradeInit`, one for each channel they would like to upgrade, again with message signer as the designated `authority` of the `IBCKeeper`

Reviewer 2: ```suggestion
If chains want to initiate the upgrade of many channels, they will need to submit a governance proposal with multiple `MsgChanUpgradeInit` messages, one for each channel they would like to upgrade, again with message signer as the designated `authority` of the `IBCKeeper`
```

Learn how to upgrade existing IBC channels.
:::

## Initializing a Channel Upgrade

> A channel upgrade is initialised by submitting the `ChanUpgradeInit` message, which can be submitted by relayer or the chain itself upon governance authorization. This message should specify an appropriate timeout window for the upgrade, in consideration of the governance method chosen (more in the section below).

Reviewer 3: ```suggestion
A channel upgrade is initialised by submitting the `ChanUpgradeInit` message, which can only be submitted by the chain itself upon governance authorization. This message should specify an appropriate timeout window for the upgrade, in consideration of the governance method chosen (more in the section below).
```

After this message is handled successfully, the channel's upgrade sequence will be incremented. This upgrade sequence will serve as a nonce for the upgrade process to avoid error receipt proof collisions.

### Governance Gating on `ChanUpgradeInit`

This can happen in several ways:

1. **A and B pre-approves certain upgrade types** : 

    A technical committee/DAO on A, elected by chain governance, initiates a governance proposal using the groups module stating their chain pre approves all channel upgrade types of the transfer module from v1 to v2. If this proposal passes, and if chain B has also pre-approved upgrades of this type (transfer v1 to v2), the channel upgrade executes in a permissioned and pre-approved manner. In this case, any relayer can submit the `MsgChanUpgradeInit` and `MsgChanUpgradeTry`.

2. **Governance-gated upgrades on A and pre-approved upgrade type on B**: 
    
    Chain A passes a governance proposal to approve an upgrade and executes `ChanUpgradeInit`. Chain B has pre-approved all upgrades of this type (initiated on A) and so any relayer can submit `MsgChanUpgradeTry` on B.

> 3. **Governance-gated upgrades on A and B**: 

Author: ah, i didnt check that it's not going to be in the first release. will update!
